SimplenoteIntents/IntentHandler.swift

```diff
@@ -2,18 +2,53 @@ import Intents
 import CoreData
 
 class IntentHandler: INExtension {
+    let coreDataManager: CoreDataManager
+    let dataController: WidgetDataController
+
+    override init() {
+        do {
+            self.coreDataManager = try CoreDataManager(StorageSettings().sharedStorageURL, for: .intents)
+            self.dataController = try WidgetDataController(coreDataManager: coreDataManager)
+        } catch {
+            fatalError()
+        }
+        super.init()
+    }
+
+
     override func handler(for intent: INIntent) -> Any {
         return self
     }
 }
 
 extension IntentHandler: NoteWidgetIntentHandling {
     func provideNoteOptionsCollection(for intent: NoteWidgetIntent, with completion: @escaping (INObjectCollection<WidgetNote>?, Error?) -> Void) {
-        var notes: [WidgetNote] = []
 
-        // TODO: add intent core data implementation.
+        // Fetch notes
+        var notes: [Note] = []
+        do {
+            notes = try dataController.notes()
+        } catch {
+            completion(nil, error)
```

Comment:
Note: regardless of whether we keep this or not, this completion runs, and then execution proceeds. This would need a `return` statement